diff --git a/node_modules/@roberto497/react-native-mapbox-navigation/android/src/main/java/com/mapboxnavigation/MapboxNavigationManager.kt b/node_modules/@roberto497/react-native-mapbox-navigation/android/src/main/java/com/mapboxnavigation/MapboxNavigationManager.kt
index 504168e..bab10e3 100644
--- a/node_modules/@roberto497/react-native-mapbox-navigation/android/src/main/java/com/mapboxnavigation/MapboxNavigationManager.kt
+++ b/node_modules/@roberto497/react-native-mapbox-navigation/android/src/main/java/com/mapboxnavigation/MapboxNavigationManager.kt
@@ -59,4 +59,9 @@ class MapboxNavigationManager(var mCallerContext: ReactApplicationContext) : Sim
     fun setShowsEndOfRouteFeedback(view: MapboxNavigationView, showsEndOfRouteFeedback: Boolean) {
         view.setShowsEndOfRouteFeedback(showsEndOfRouteFeedback)
     }
+
+    @ReactProp(name = "exclude")
+    fun setExclude(view: MapboxNavigationView, exclude: String?) {
+        view.setExclude(exclude)
+    }
 }
diff --git a/node_modules/@roberto497/react-native-mapbox-navigation/android/src/main/java/com/mapboxnavigation/MapboxNavigationView.kt b/node_modules/@roberto497/react-native-mapbox-navigation/android/src/main/java/com/mapboxnavigation/MapboxNavigationView.kt
index 519c58c..3a893b9 100644
--- a/node_modules/@roberto497/react-native-mapbox-navigation/android/src/main/java/com/mapboxnavigation/MapboxNavigationView.kt
+++ b/node_modules/@roberto497/react-native-mapbox-navigation/android/src/main/java/com/mapboxnavigation/MapboxNavigationView.kt
@@ -54,6 +54,7 @@ import com.mapbox.navigation.tripdata.progress.model.EstimatedTimeToArrivalForma
 import com.mapbox.navigation.tripdata.progress.model.PercentDistanceTraveledFormatter
 import com.mapbox.navigation.tripdata.progress.model.TimeRemainingFormatter
 import com.mapbox.navigation.tripdata.progress.model.TripProgressUpdateFormatter
+import com.mapbox.navigation.ui.maps.NavigationStyles
 import com.mapbox.navigation.ui.maps.camera.NavigationCamera
 import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource
 import com.mapbox.navigation.ui.maps.camera.lifecycle.NavigationBasicGesturesHandler
@@ -81,6 +82,7 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
     private var arrivalNotificationHasDisplayed = false
 
     private var destination: Point? = null
+    private var exclude: String? = null
     private var showsEndOfRouteFeedback = false
 
     /**
@@ -241,10 +243,13 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
         viewportDataSource.evaluate()
 
         // draw the upcoming maneuver arrow on the map
-        val style = binding.mapView.mapboxMap.style
-        if (style != null) {
+        binding.mapView.mapboxMap.getStyle { style ->
             val maneuverArrowResult = routeArrowApi.addUpcomingManeuverArrow(routeProgress)
             routeArrowView.renderManeuverUpdate(style, maneuverArrowResult)
+
+            routeLineApi.updateWithRouteProgress(routeProgress) { result ->
+                routeLineView.renderRouteLineUpdate(style, result)
+            }
         }
 
         // update top banner with maneuver instructions
@@ -302,8 +307,8 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
             routeLineApi.setNavigationRoutes(
                 routeUpdateResult.navigationRoutes
             ) { value ->
-                binding.mapView.mapboxMap.style?.apply {
-                    routeLineView.renderRouteDrawData(this, value)
+                binding.mapView.mapboxMap.getStyle { style ->
+                    routeLineView.renderRouteDrawData(style, value)
                 }
             }
 
@@ -312,8 +317,7 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
             viewportDataSource.evaluate()
         } else {
             // remove the route line and route arrow from the map
-            val style = binding.mapView.mapboxMap.style
-            if (style != null) {
+            binding.mapView.mapboxMap.getStyle { style ->
                 routeLineApi.clearRouteLine { value ->
                     routeLineView.renderClearRouteLineValue(
                         style,
@@ -339,6 +343,11 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
         post(measureAndLayout)
     }
 
+    override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
+        super.onSizeChanged(w, h, oldw, oldh)
+        post(measureAndLayout)
+    }
+
     private val measureAndLayout = Runnable {
         measure(
             MeasureSpec.makeMeasureSpec(width, MeasureSpec.EXACTLY),
@@ -443,15 +452,10 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
                 .build()
         )
 
-        // load map style
-        binding.mapView.scalebar.enabled = false
-        binding.mapView.compass.enabled = false
-        binding.mapView.mapboxMap.loadStyle(Style.MAPBOX_STREETS)
-
         // configure Route line options
         val mapboxRouteLineApiOptions = MapboxRouteLineApiOptions.Builder().build()
         val mapboxRouteLineViewOptions = MapboxRouteLineViewOptions.Builder(context)
-            .routeLineBelowLayerId("road-label")
+            .routeLineBelowLayerId("road-label-navigation")
             .build()
         val mapboxRouteArrowOptions = RouteArrowOptions.Builder(context).build()
 
@@ -459,6 +463,13 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
         routeLineView = MapboxRouteLineView(mapboxRouteLineViewOptions)
         routeArrowView = MapboxRouteArrowView(mapboxRouteArrowOptions)
 
+        // load map style
+        binding.mapView.scalebar.enabled = false
+        binding.mapView.compass.enabled = false
+        binding.mapView.mapboxMap.loadStyle(NavigationStyles.NAVIGATION_DAY_STYLE) { style ->
+            routeLineView.initializeLayers(style)
+        }
+
         // initialize view interactions
         binding.stop.setOnClickListener {
             clearRouteAndStopNavigation()
@@ -481,7 +492,7 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
                 .getJSModule(RCTEventEmitter::class.java)
                 .receiveEvent(id, "onChat", event)
         }
-        binding.startNavigation.text = "Iniciar navegação";
+        binding.startNavigation.text = "Start Navigation";
         binding.startNavigation.setOnClickListener {
             binding.startNavigation.visibility = View.GONE
             this.destination?.let { it ->  findRoute(it) }
@@ -550,15 +561,20 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
         } ?: return
 
         try {
+            val routeOptionsBuilder = RouteOptions.builder()
+                .applyDefaultNavigationOptions()
+                .applyLanguageAndVoiceUnitOptions(context)
+                .coordinatesList(listOf(originPoint, destination))
+                .profile(DirectionsCriteria.PROFILE_DRIVING)
+                .steps(true)
+                .language("en")
+
+            if (!exclude.isNullOrEmpty()) {
+                routeOptionsBuilder.exclude(exclude)
+            }
+
             mapboxNavigation.requestRoutes(
-                RouteOptions.builder()
-                    .applyDefaultNavigationOptions()
-                    .applyLanguageAndVoiceUnitOptions(context)
-                    .coordinatesList(listOf(originPoint, destination))
-                    .profile(DirectionsCriteria.PROFILE_DRIVING)
-                    .steps(true)
-                    .language("pt-BR")
-                    .build(),
+                routeOptionsBuilder.build(),
                 object : NavigationRouterCallback {
                     override fun onRoutesReady(
                         routes: List<NavigationRoute>,
@@ -626,6 +642,10 @@ class MapboxNavigationView(private val context: ThemedReactContext) :
         this.destination = destination
     }
 
+    fun setExclude(exclude: String?) {
+        this.exclude = exclude
+    }
+
     fun setShowsEndOfRouteFeedback(showsEndOfRouteFeedback: Boolean) {
         this.showsEndOfRouteFeedback = showsEndOfRouteFeedback
     }
diff --git a/node_modules/@roberto497/react-native-mapbox-navigation/src/typings.ts b/node_modules/@roberto497/react-native-mapbox-navigation/src/typings.ts
index 566ba96..0b6042c 100644
--- a/node_modules/@roberto497/react-native-mapbox-navigation/src/typings.ts
+++ b/node_modules/@roberto497/react-native-mapbox-navigation/src/typings.ts
@@ -32,4 +32,5 @@ export interface IMapboxNavigationProps {
   onChat?: () => void;
   showsEndOfRouteFeedback?: boolean;
   hideStatusView?: boolean;
+  exclude?: string;
 }
